### PR TITLE
Global classes in core CSS

### PR DESCRIPTION
**Pull request by raghunayyar**
A small start to shorten the CSS we write while building both apps or core. I think we should have a bunch of classes that we necessarily need, declared in core that can be re-used. Though, I can introduce a separate file for that but this seems good for the starters. What do you guys think? @jancborchardt @MorrisJobke @owncloud/designers

### CODE REVIEW

File: core/css/styles.css [hunk 1]
```diff
@@ -24,6 +24,18 @@ body {
 	height: auto;
 }
 
+.pull-left {
+	float: left;
+}
+
+.pull-right {
+	float: right;
+}
```
Comment (PVince81) [inline core/css/styles.css]: also add clear-left and clear-right and/or clear-both.
This can be needed when containers contain floats and need to be reset to make sure it gets a proper height (happened to me in the version tab PR)

File: core/css/styles.css [hunk 2]
```diff
@@ -24,6 +24,18 @@ body {
...
+
+.hide {
```
Comment (PVince81) [inline core/css/styles.css]: we already have a ".hidden" class.

I'd suggest you rather go through the existing classes and move them here

Comment (raghunayyar) [inline core/css/styles.css]: Sure, just that I couldn't find that or any documentation.. Will do that asap.

**Comment by raghunayyar** [comment]
@PVince81 I pushed a few more on top, the ones I could find.

**Comment by MorrisJobke** [comment]
There is the login button fix commit in here. Please delete it from this PR.

Also a separate class would be nice.

**Comment by raghunayyar** [comment]
Damn, missed that. I will remove that asap.

File: core/css/styles.css [hunk 3]
```diff
@@ -461,7 +487,7 @@ input[type="submit"].enabled {
 #body-login #submit.login {
 	position: absolute;
 	right: 0;
-	top: 49px;
```
Comment (jancborchardt) [inline core/css/styles.css]: This is not part of this pull request, no?

Comment (MorrisJobke) [inline core/css/styles.css]: > This is not part of this pull request, no?

Yep. I already mentioned it ;)

File: core/css/styles.css (hunk 1 above)
Comment (raghunayyar) [inline core/css/styles.css]: done. @PVince81

**Comment by raghunayyar** [comment]
@jancborchardt @PVince81 added a couple of global glasses, you can suggest more, but I think this is a good starting point.

**Comment by MorrisJobke** [comment]
@owncloud/designers Please review

**Comment by jancborchardt** [comment]
Can we move these out into a separate CSS file to not add more to styles.css? Could be named »reusable.css« or something like that.

**Comment by raghunayyar** [comment]
Sure, I suggested that at the conf too. :P

**Comment by jancborchardt** [comment]
Great, then go for it :)

**Comment by PVince81** [comment]
How about `global.css` instead ?
In an ideal world, almost all CSS should be reusable.

**Comment by raghunayyar** [comment]
@PVince81 named it global.css only. Can you tell me where to add this css file in PHP templates?

**Comment by jancborchardt** [comment]
global.css is fine for me as well. Maybe the reset should be moved there as well then, and this file should be first in the list of loaded CSS files?

Also, @raghunayyar when you are finished, can you squash the commits down to one.

**Comment by MorrisJobke** [comment]
> Can you tell me where to add this css file in PHP templates?

https://github.com/owncloud/core/blob/053effaa5124ac527712ac97193eb444259c69bc/lib/private/template.php#L106-L116

**Comment by raghunayyar** [comment]
@MorrisJobke @jancborchardt @PVince81 please check now. :)

**Comment by raghunayyar** [comment]
@jancborchardt or @PVince81 can you please squash my commits, I screwed up the last time I tried doing that.

**Comment by jancborchardt** [comment]
@raghunayyar it’s pretty simple, just follow https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/

**Comment by raghunayyar** [comment]
@jancborchardt thanks a lot for the resource. The PR is ready now. ^_^

**Comment by MorrisJobke** [comment]
Tested, changes looks good and it still works :+1:

**Comment by MorrisJobke** [comment]
@owncloud/designers Easy to review ;)

cc @Henni

**Comment by Henni** [comment]
:+1:

**Comment by PVince81** [comment]
Regression: https://github.com/owncloud/core/issues/20566
